src/test/resources/regressions/features/defunc/mutex-and-waitgroup1.gobra

```diff
@@ -10,84 +10,142 @@ type Accum struct {
 	m *sync.Mutex
 }
 
-pred AccInv(x *int) {
-	acc(x)
+pred AccInv(x *int, seenSlice []int) {
+	acc(x) && (forall i int :: 0 <= i && i < len(seenSlice) ==> (acc(&seenSlice[i], 1/2))) && Sum(seenSlice) == *x
 }
 
-// TODO: debug this
-// requires acc(c) && acc(c.sum) && acc(c.m) && *(c.m) == sync.Mutex{}
+// TODO: change `c.m.state == 0 && c.m.stema == 0` to *(c.m) == sync.Mutex{} after
+//       fixing issue #190
 requires acc(c) && acc(c.sum) && acc(c.m) && c.m.state == 0 && c.m.stema == 0
-requires c.sum != nil
-ensures acc(c) && c.m.LockP() && c.m.LockInv() == (AccInv!<c.sum!>)
-func (c *Accum) Init() {
-	fold AccInv!<c.sum!>()
-	c.m.SetInv(AccInv!<c.sum!>)
+requires *c.sum == 0
+requires forall i int :: 0 <= i && i < len(seenSlice) ==> acc(&seenSlice[i], 1/2)
+requires forall i int :: 0 <= i && i < len(seenSlice) ==> seenSlice[i] == 0
+ensures acc(c) && c.m.LockP() && c.m.LockInv() == (AccInv!<c.sum, seenSlice!>)
+func (c *Accum) Init(ghost seenSlice []int) {
```

Comment:
Would it be possible to make this function ghost? As it only performs ghost operations, ghost erasure would produce more meaningful LOCs if this is not erased to an empty function

Reply:
Yes, you are right! There's no actual code here, only ghost code